strictdoc/server/routers/main_router.py

```diff
@@ -2696,6 +2698,49 @@ def get_search(q: Optional[str] = None):
             status_code=200,
         )
 
+    # @relation(SDOC-SRS-120, scope=function)
+    @router.get("/autocomplete/uid", response_class=Response)
+    def get_autocomplete_results(q: Optional[str] = None):
+        output = ""
+        if q is not None:
+            query_words = q.lower().split()
+            resulting_nodes = []
+            for (
+                document
+            ) in export_action.traceability_index.document_tree.document_list:
+                document_iterator = (
+                    export_action.traceability_index.get_document_iterator(
+                        document
+                    )
+                )
+                for node in document_iterator.all_content(
+                    print_fragments=False, print_fragments_from_files=False
+                ):
+                    if node.reserved_uid is not None:
+                        words = node.reserved_uid.strip().lower()
+                        if node.reserved_title is not None:
+                            words = (
+                                words
+                                + " "
+                                + node.reserved_title.strip().lower()
+                            )
+                        if all(word.lower() in words for word in query_words):
```

Comment:
1) I believe you don't need lower() here because the query_words have already been lowered above with `query_words = q.lower().split()`.

2) StrictDoc has a recently adopted convention that the temporary variables such as those used inside small for loops or generator expressions are suffixed with _:

```py
if all(word_ in words for word_ in query_words)
```

this helps to better visualize that a given variable is only active within a very small scope.